src/zenml/cli/utils.py

```diff
@@ -210,15 +206,40 @@ def __dictify(model: M) -> Dict[str, str]:
         Returns:
             Dict of model attributes.
         """
-        items = (
-            {
-                key: str(value)
-                for key, value in model.dict().items()
-                if key not in exclude_columns
-            }
-            if columns is None
-            else {key: str(model.dict()[key]) for key in columns}
-        )
+        # Explicitly defined columns take precedence over exclude columns
+        if not columns:
+            include_columns = [
+                k for k in model.dict().keys() if k not in exclude_columns
+            ]
+        else:
+            include_columns = columns
+
+        items: Dict[str, Any] = {}
+
+        for k in include_columns:
+            value = getattr(model, k)
+            # In case the response model contains nested BaseResponseModels
+            #  we want to attempt to represent them by name, if they contain
+            #  such a field, else the id is used
+            if isinstance(value, BaseResponseModel):
+                if "name" in value.__fields__:
+                    items[k] = str(value.name)  # type: ignore[attr-defined]
+                else:
+                    items[k] = str(value.id)
+
+            # If it is a list of BaseResponseModels access each Model within
+            #  the list and extract either name or id
+            elif isinstance(value, list) and issubclass(
+                model.__fields__[k].type_, BaseResponseModel
+            ):
```

Comment:
Not sure why the two statements are different. Perhaps I can tag @AlexejPenner here. :)

Reply:
@fa9r  Alright, this is supposed to catch nested models:

isinstance(value, BaseResponseModel) this will catch simple nested models:

UserResponseModel.role is type RoleResponseModel:
isinstance(UserResponseModel.role, BaseResponseModel) == True

issubclass(model.__fields__[runs].type_, BaseResponseModel) will catch List of models that can be attributes of the top-level model

PipelineResponseModel.runs is type List[PipelineRunsResponseModel]:
isinstance(PipelineResponseModel.runs, BaseResponseModel) == False
isinstance(PipelineResponseModel.runs, list) == True
issubclass(model.__fields__[runs].type_, BaseResponseModel) == True

Reply:
@AlexejPenner I don't fully get your answer, but one thing I noticed is in your response you gave examples of

```issubclass(model.__fields__[runs].type_, BaseResponseModel)```

but in code you have

```issubclass(model.__fields__[k].type_, BaseResponseModel)```

Reply:
Yeah that was on purpose to be more accurate for the example I was giving. I'll try coming up with a better explanation